### ISSUE TITLE
Huge time consuming during database initialization

### ISSUE THREAD

**Issue by loveyana** [spiffe/spire#4430]
* **Version**:  v1.6.3 
* **Platform**: linux
* **Subsystem**: datastore

In `https://github.com/spiffe/spire/pull/3941` introduced the logic of data cleaning during the initialization phase.

But when executing the DELETE statement `DELETE FROM node_resolver_map_entries WHERE spiffe_id NOT IN (SELECT spiffe_id FROM attested_node_entries)` with a very large amount of data, it can take a long time, about 20 minutes for us.

Mainly due to the following two reasons:
1. At present, we attested_node_entries data table data volume reaches about 40w, node_resolver_map_entries data table data volume is about 500w, which will cause the subquery to take a long time to execute.
2. Since spire-spire is usually deployed in the form of deployment or statefulset in the kubernetes cluster, there will be multiple instances at the same time, so the data table will be locked when a single instance restarts the DELETE operation. When `Agent Attest` requests appear in other instances, lock conflicts will occur, resulting in prolonged execution time.

At the same time, because the spire-server Pod usually has Kubernetes probes, we need it to be ready within 30 seconds, but due to the long DELETE operation, the restart operation of the spire-server in the cluster continues to fail. 

In the case of deploying spire-server in the cluster, multiple DELETE operations in parallel will also lead to excessive database pressure.

At present, I have deleted the cleaning logic in the initialization stage in the internal version. I also hope that the community will take into account the data scale and deployment in the production environment when performing similar MR, and do not perform such operations in the initialization stage.

**Comment by azdagron** [comment]
Thanks for opening this, @loveyana. This was very enlightening. Thanks for bringing it to our attention. It will help us make better decisions when adding these kinds of "startup" checks in the future.

As an aside, the data cleanup is no longer needed for 1.8.0 and has been removed.

Thanks again ❤️
